src/generic_containers.rs

```diff
@@ -0,0 +1,114 @@
+//! **Private** generalizations of containers:
+//! - `Map`: `BTreeMap` and `HashMap` (any hasher).
+
+#![cfg(feature = "use_alloc")]
+
+use alloc::collections::BTreeMap;
+use alloc::vec::Vec;
+#[cfg(feature = "use_std")]
+use core::hash::{BuildHasher, Hash};
+#[cfg(feature = "use_std")]
+use std::collections::HashMap;
+
+pub trait Map {
```

Comment:
If we want to commit to the "pass an existing map to extend" pattern then you could add an implementation of `Map` for `&mut M where M: Map` to make it even more flexible.

Reply:
I find this interesting and I will keep it in mind but I want to use `Map` on other things than `GroupingMap` so I'd rather keep it as simple as possible for now.